app/src/main/java/com/thanhclub/musicdemo/base/BaseSongFragment.java

```diff
@@ -19,30 +20,39 @@
 import com.thanhclub.musicdemo.manager.DatabaseManager;
 import com.thanhclub.musicdemo.manager.SongManager;
 import com.thanhclub.musicdemo.model.Song;
+import com.thanhclub.musicdemo.services.PlaySongService;
 
 import java.util.ArrayList;
-import java.util.List;
-
 
 public abstract class BaseSongFragment extends Fragment implements OnItemClickListener, OnLongItemClickListener {
     protected View view;
     protected RecyclerView recycler;
     protected SongAdpater songAdpater;
     protected RecyclerView.LayoutManager layoutManager;
     protected SongManager songManager;
-    protected List<Song> songs;
+    protected ArrayList<Song> songs;
     protected DatabaseManager databaseManager;
+    protected PlaySongService service;
+    public static final String ACTION_START_SERVICE = "ACTION_START_SERVICE";
```

Comment:
2.8 Class member ordering
This is not the only correct solution, but it is recommended to use the following order :

Constants
Fields
Constructors
Override methods and callbacks (public or private)
Public methods
Private methods
Inner classes or interfaces